In [2]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore") 

---

###
###

# Hoja 1

Vamos a dar una vista rapida de los datos

In [180]:
Acc_vel_loc_sinrangos = pd.read_csv("../../data/converted/Acc_vel_loc_sinrangos.csv")
print(Acc_vel_loc_sinrangos.info())
Acc_vel_loc_sinrangos.head(20)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3065 entries, 0 to 3064
Columns: 126 entries, Unnamed: 0 to 10000 Mbps
dtypes: float64(121), int64(1), object(4)
memory usage: 2.9+ MB
None


Unnamed: 0     Provincia        Partido  \
0            0  BUENOS AIRES     25 de Mayo   
1            1  BUENOS AIRES     25 de Mayo   
2            2  BUENOS AIRES     25 de Mayo   
3            3  BUENOS AIRES     25 de Mayo   
4            4  BUENOS AIRES     25 de Mayo   
5            5  BUENOS AIRES     25 de Mayo   
6            6  BUENOS AIRES     25 de Mayo   
7            7  BUENOS AIRES     25 de Mayo   
8            8  BUENOS AIRES     25 de Mayo   
9            9  BUENOS AIRES     25 de Mayo   
10          10  BUENOS AIRES     9 de Julio   
11          11  BUENOS AIRES     9 de Julio   
12          12  BUENOS AIRES     9 de Julio   
13          13  BUENOS AIRES     9 de Julio   
14          14  BUENOS AIRES     9 de Julio   
15          15  BUENOS AIRES     9 de Julio   
16          16  BUENOS AIRES     9 de Julio   
17          17  BUENOS AIRES     9 de Julio   
18          18  BUENOS AIRES  Adolfo Alsina   
19          19  BUENOS AIRES  Adolfo Alsina   

                                        Localidad Link Indec  Otros  \
0                                      25 de Mayo    6854100    NaN   
1                                 Agustín Mosconi    6854010    NaN   
2                                       Del Valle    6854020    NaN   
3                                       Ernestina    6854030    NaN   
4                               Gobernador Ugarte    6854040    NaN   
5                                Lucas Monteverde    6854050    NaN   
6                          Norberto de la Riestra    6854060    NaN   
7                                      Pedernales    6854070    NaN   
8                                     San Enrique    6854080    NaN   
9                                          Valdés    6854090    NaN   
10                                  12 de Octubre    6588030    NaN   
11                                     9 de Julio    6588100    NaN   
12       Alfredo Demarchi  (Est. Facundo Quiroga)    6588010    NaN   
13                                       Dudignac    6588040    NaN   
14                Manuel B. Gonnet  (Est. French)    6588060    NaN   
15                                          Morea    6588080    NaN   
16                                      Patricios    6588110    NaN   
17  Villa General Fournier  (Est. 9 de Julio Sud)    6588120    NaN   
18                                         Carhué    6007010    NaN   
19                    Colonia San Miguel Arcángel    6007020    NaN   

    0,256 Mbps  0,5 Mbps  0,512 Mbps  0,75 Mbps  ...  680 Mbps  700 Mbps  \
0          NaN       2.0         NaN       19.0  ...       NaN       NaN   
1          NaN       NaN         NaN        NaN  ...       NaN       NaN   
2          NaN       1.0         NaN        NaN  ...       NaN       NaN   
3          NaN       NaN         NaN        NaN  ...       NaN       NaN   
4          NaN       NaN         NaN        NaN  ...       NaN       NaN   
5          NaN       NaN         NaN        NaN  ...       NaN       NaN   
6          NaN       NaN         NaN        NaN  ...       NaN       NaN   
7          NaN       NaN         NaN        NaN  ...       NaN       NaN   
8          NaN       NaN         NaN        NaN  ...       NaN       NaN   
9          NaN       NaN         NaN        NaN  ...       NaN       NaN   
10         NaN       NaN         NaN        NaN  ...       NaN       NaN   
11         NaN       NaN         NaN        9.0  ...       NaN       NaN   
12         NaN       NaN         NaN        NaN  ...       NaN       NaN   
13         NaN       NaN         NaN        NaN  ...       NaN       NaN   
14         NaN       NaN         NaN        NaN  ...       NaN       NaN   
15         NaN       NaN         NaN        NaN  ...       NaN       NaN   
16         NaN       NaN         NaN        NaN  ...       NaN       NaN   
17         NaN       NaN         NaN        NaN  ...       NaN       NaN   
18         NaN       NaN         NaN        NaN  ...       NaN       NaN   
19  

In [181]:
Acc_vel_loc_sinrangos.describe()

Unnamed: 0         Otros  0,256 Mbps      0,5 Mbps  0,512 Mbps  \
count  3065.000000   1030.000000   36.000000    642.000000  525.000000   
mean   1532.000000    318.614563    2.944444     42.978193    2.100952   
std     884.933613   2964.939611    5.806127    985.466627    4.684645   
min       0.000000    -15.000000    1.000000      0.000000    1.000000   
25%     766.000000      1.000000    1.000000      1.000000    1.000000   
50%    1532.000000      6.000000    1.000000      1.000000    1.000000   
75%    2298.000000     48.000000    2.000000      2.000000    1.000000   
max    3064.000000  87722.000000   31.000000  24968.000000   61.000000   

         0,75 Mbps       1 Mbps     1,5 Mbps       2 Mbps  2,2 Mbps  ...  \
count   284.000000   872.000000    15.000000   677.000000       1.0  ...   
mean     39.588028    66.642202   132.533333   101.428360      18.0  ...   
std     118.563946   222.345674   291.222268   278.821468       NaN  ...   
min       1.000000     0.000000     7.000000     1.000000      18.0  ...   
25%       3.000000     2.000000    31.000000     4.000000      18.0  ...   
50%       9.500000    10.000000    47.000000    21.000000      18.0  ...   
75%      28.000000    40.250000    75.000000    77.000000      18.0  ...   
max    1307.000000  4111.000000  1170.000000  3530.000000      18.0  ...   

       680 Mbps  700 Mbps    800 Mbps  850 Mbps  900 Mbps   999 Mbps  \
count       1.0       4.0    4.000000       1.0       1.0  17.000000   
mean        5.0       1.0  153.500000       1.0     110.0   3.352941   
std         NaN       0.0  273.381906       NaN       NaN   5.700232   
min         5.0       1.0    1.000000       1.0     110.0   1.000000   
25%         5.0       1.0   10.750000       1.0     110.0   1.000000   
50%         5.0       1.0   25.000000       1.0     110.0   2.000000   
75%         5.0       1.0  167.750000       1.0     110.0   3.000000   
max         5.0       1.0  563.000000       1.0     110.0  25.000000   

          1000 Mbps  1024 Mbps  4000 Mbps  10000 Mbps  
count    420.000000        2.0        1.0    2.000000  
mean     609.683333        1.0        1.0    5.000000  
std     3431.387543        0.0        NaN    5.656854  
min        1.000000        1.0        1.0    1.000000  
25%       17.000000        1.0        1.0    3.000000  
50%      139.500000        1.0        1.0    5.000000  
75%      498.000000        1.0        1.0    7.000000  
max    68405.000000        1.0        1.0    9.000000  

[8 rows x 122 columns]

Vamos a rellenar esos datos nulos para que al menos podamos realizar la suma de conexiones totales por cada rango de servicio contratado

In [182]:
Acc_vel_loc_sinrangos = Acc_vel_loc_sinrangos.fillna(0)

In [183]:
Acc_vel_loc_sinrangos

Unnamed: 0     Provincia      Partido  \
0              0  BUENOS AIRES   25 de Mayo   
1              1  BUENOS AIRES   25 de Mayo   
2              2  BUENOS AIRES   25 de Mayo   
3              3  BUENOS AIRES   25 de Mayo   
4              4  BUENOS AIRES   25 de Mayo   
...          ...           ...          ...   
3060        3060       TUCUMAN   Tafí Viejo   
3061        3061       TUCUMAN      Trancas   
3062        3062       TUCUMAN      Trancas   
3063        3063       TUCUMAN  Yerba Buena   
3064        3064       TUCUMAN  Yerba Buena   

                               Localidad Link Indec  Otros  0,256 Mbps  \
0                             25 de Mayo    6854100    0.0         0.0   
1                        Agustín Mosconi    6854010    0.0         0.0   
2                              Del Valle    6854020    0.0         0.0   
3                              Ernestina    6854030    0.0         0.0   
4                      Gobernador Ugarte    6854040    0.0         0.0   
...                                  ...        ...    ...         ...   
3060  Villa Mariano Moreno - El Colmenar   90105100    0.0         0.0   
3061                San Pedro de Colalao   90112020    0.0         0.0   
3062                   Villa  de Trancas   90112030    0.0         0.0   
3063                       Villa Carmela   90119020    8.0         0.0   
3064            Yerba Buena - Marcos Paz   90119030  176.0         0.0   

      0,5 Mbps  0,512 Mbps  0,75 Mbps  ...  680 Mbps  700 Mbps  800 Mbps  \
0          2.0         0.0       19.0  ...       0.0       0.0       0.0   
1          0.0         0.0        0.0  ...       0.0       0.0       0.0   
2          1.0         0.0        0.0  ...       0.0       0.0       0.0   
3          0.0         0.0        0.0  ...       0.0       0.0       0.0   
4          0.0         0.0        0.0  ...       0.0       0.0       0.0   
...        ...         ...        ...  ...       ...       ...       ...   
3060       0.0         0.0        0.0  ...       0.0       0.0       0.0   
3061       0.0         0.0        0.0  ...       0.0       0.0       0.0   
3062       0.0         0.0        0.0  ...       0.0       0.0       0.0   
3063       0.0         0.0        0.0  ...       0.0       0.0       0.0   
3064       0.0         0.0        0.0  ...       0.0       0.0       0.0   

      850 Mbps  900 Mbps  999 Mbps  1000 Mbps  1024 Mbps  4000 Mbps  \
0          0.0       0.0       0.0        0.0        0.0        0.0   
1          0.0       0.0       0.0        0.0        0.0        0.0   
2          0.0       0.0       0.0        0.0        0.0        0.0   
3          0.0       0.0       0.0        0.0        0.0        0.0   
4          0.0       0.0       0.0        0.0        0.0        0.0   
...        ...       ...       ...        ...        ...        ...   
3060       0.0       0.0       0.0        5.0        0.0        0.0   
3061       0.0       0.0       0.0        0.0        0.0        0.0   
3062       0.0       0.0       0.0        0.0        0.0        0.0   
3063       0.0       0.0       0.0        0.0        0.0        0.0   
3064       0.0       0.0       0.0       20.0        0.0        0.0   

      10000 Mbps  
0            0.0  
1            0.0  
2            0.0  
3            0.0  
4            0.0  
...          ...  
3060         0.0  
3061         0.0  
3062         0.0  
3063         0.0  
3064         0.0  

[3065 rows x 126 columns]

In [184]:
Acc_vel_loc_sinrangos['Link Indec'].value_counts()

Link Indec
Sin Datos    74
6441030      27
6371010      27
6427010      15
6840010      15
             ..
14140110      1
14140150      1
14140160      1
14140170      1
14140010      1
Name: count, Length: 2664, dtype: int64

Voy a mirar si hay algunas columnas con datos, donde 'Link Indec' aparece como 'sin datos', caso contrario elimino las filas  
- Con describe voy a ver muy por encima si encuentro algun dato anormal

In [185]:
link_indec = Acc_vel_loc_sinrangos[Acc_vel_loc_sinrangos['Link Indec']=='Sin Datos']
link_indec.describe()

Unnamed: 0         Otros  0,256 Mbps      0,5 Mbps  0,512 Mbps  \
count    74.000000     74.000000        74.0     74.000000        74.0   
mean   1605.837838   1756.162162         0.0    339.094595         0.0   
std     884.946435  10269.378169         0.0   2902.276344         0.0   
min      43.000000    -15.000000         0.0      0.000000         0.0   
25%     852.750000      0.000000         0.0      0.000000         0.0   
50%    1613.500000      0.000000         0.0      0.000000         0.0   
75%    2323.500000     56.250000         0.0      1.000000         0.0   
max    3042.000000  87722.000000         0.0  24968.000000         0.0   

       0,75 Mbps     1 Mbps  1,5 Mbps     2 Mbps  2,2 Mbps  ...  680 Mbps  \
count  74.000000  74.000000      74.0  74.000000      74.0  ...      74.0   
mean    0.067568   2.283784       0.0   2.702703       0.0  ...       0.0   
std     0.477754  10.552569       0.0  10.169427       0.0  ...       0.0   
min     0.000000   0.000000       0.0   0.000000       0.0  ...       0.0   
25%     0.000000   0.000000       0.0   0.000000       0.0  ...       0.0   
50%     0.000000   0.000000       0.0   0.000000       0.0  ...       0.0   
75%     0.000000   0.000000       0.0   0.000000       0.0  ...       0.0   
max     4.000000  75.000000       0.0  62.000000       0.0  ...       0.0   

       700 Mbps  800 Mbps  850 Mbps  900 Mbps  999 Mbps    1000 Mbps  \
count      74.0      74.0      74.0      74.0      74.0    74.000000   
mean        0.0       0.0       0.0       0.0       0.0    67.567568   
std         0.0       0.0       0.0       0.0       0.0   581.238194   
min         0.0       0.0       0.0       0.0       0.0     0.000000   
25%         0.0       0.0       0.0       0.0       0.0     0.000000   
50%         0.0       0.0       0.0       0.0       0.0     0.000000   
75%         0.0       0.0       0.0       0.0       0.0     0.000000   
max         0.0       0.0       0.0       0.0       0.0  5000.000000   

       1024 Mbps  4000 Mbps  10000 Mbps  
count  74.000000       74.0        74.0  
mean    0.013514        0.0         0.0  
std     0.116248        0.0         0.0  
min     0.000000        0.0         0.0  
25%     0.000000        0.0         0.0  
50%     0.000000        0.0         0.0  
75%     0.000000        0.0         0.0  
max     1.000000        0.0         0.0  

[8 rows x 122 columns]

In [186]:
Acc_vel_loc_sinrangos['Unnamed: 0'].value_counts()

Unnamed: 0
3064    1
0       1
1       1
2       1
3       1
       ..
28      1
29      1
30      1
31      1
32      1
Name: count, Length: 3065, dtype: int64

La columna 'Unnamed: 0', tiene valores faltantes, son 3065 filas y aparece una longitud de 2995, al considerarla una columna poco util la elimino directamente

In [187]:
Acc_vel_loc_sinrangos = Acc_vel_loc_sinrangos.drop('Unnamed: 0', axis=1)

In [188]:
Acc_vel_loc_sinrangos.Localidad.value_counts()

Localidad
Otros                      70
San Pedro                   8
San Antonio                 6
San José                    6
Santa Rosa                  5
                           ..
Copacabana                  1
Deán Funes                  1
Esquina del Alambre         1
Olivares de San Nicolás     1
La Playosa                  1
Name: count, Length: 2778, dtype: int64

Considero que al no conocer la ubicacion de la localidad, no ayudaria al analisis de la ubicacion y el servicio que se requiere en el lugar.  
La voy a eliminar para no tener una agrupacion erronea de las localidades

In [189]:
Acc_vel_loc_sinrangos = Acc_vel_loc_sinrangos[Acc_vel_loc_sinrangos.Localidad!='Otros']
Acc_vel_loc_sinrangos.Localidad.value_counts()

Localidad
San Pedro                  8
San Antonio                6
San José                   6
Santa Rosa                 5
San Miguel                 5
                          ..
Deán Funes                 1
Esquina del Alambre        1
Olivares de San Nicolás    1
Quilino                    1
Pasco                      1
Name: count, Length: 2777, dtype: int64

cabe mencionar que hay localidades que coinciden con el nombre de otras localidades ubicadas en otras provincias.  
- por ejemplo:

In [190]:
san_jose = Acc_vel_loc_sinrangos[Acc_vel_loc_sinrangos.Localidad=='San José']
san_jose

Provincia         Partido Localidad Link Indec   Otros  0,256 Mbps  \
177   BUENOS AIRES  Coronel Suárez  San José    6203070     9.0         0.0   
776      CATAMARCA     Santa María  San José   10091140     0.0         0.0   
1199       CORDOBA      San Javier  San José   14133180     0.0         0.0   
1381    ENTRE RIOS           Colón  San José   30008090  1070.0         0.0   
1949       MENDOZA       Tupungato  San José   50126050     0.0         0.0   
1960      MISIONES       Apóstoles  San José   54007060     0.0         0.0   

      0,5 Mbps  0,512 Mbps  0,75 Mbps  1 Mbps  ...  680 Mbps  700 Mbps  \
177        0.0         0.0        0.0     0.0  ...       0.0       0.0   
776        0.0         1.0        0.0     1.0  ...       0.0       0.0   
1199       0.0         0.0        0.0     0.0  ...       0.0       0.0   
1381       0.0         0.0        0.0     9.0  ...       0.0       0.0   
1949       0.0         0.0        0.0     0.0  ...       0.0       0.0   
1960       0.0         0.0        0.0   424.0  ...       0.0       0.0   

      800 Mbps  850 Mbps  900 Mbps  999 Mbps  1000 Mbps  1024 Mbps  4000 Mbps  \
177        0.0       0.0       0.0       0.0        0.0        0.0        0.0   
776        0.0       0.0       0.0       0.0        0.0        0.0        0.0   
1199       0.0       0.0       0.0       0.0        0.0        0.0        0.0   
1381       0.0       0.0       0.0       0.0        0.0        0.0        0.0   
1949       0.0       0.0       0.0       0.0        0.0        0.0        0.0   
1960       0.0       0.0       0.0       0.0        0.0        0.0        0.0   

      10000 Mbps  
177          0.0  
776          0.0  
1199         0.0  
1381         0.0  
1949         0.0  
1960         0.0  

[6 rows x 125 columns]

###
###

Terminamos de analizar las otras columnas categoricas

In [191]:
Acc_vel_loc_sinrangos.Partido.value_counts()

Partido
San Justo             55
General San Martín    42
Castellanos           38
Colón                 36
San Martín            34
                      ..
Hipólito Yrigoyen      1
Angaco                 1
Zonda                  1
Ullum                  1
Magallanes             1
Name: count, Length: 430, dtype: int64

In [192]:
Acc_vel_loc_sinrangos.Provincia.value_counts()

Provincia
BUENOS AIRES           700
CORDOBA                378
SANTA FE               351
ENTRE RIOS             137
SANTIAGO DEL ESTERO    135
MENDOZA                131
SALTA                  120
RIO NEGRO               93
SAN LUIS                92
MISIONES                90
JUJUY                   86
LA PAMPA                81
CATAMARCA               74
CHACO                   73
LA RIOJA                68
TUCUMAN                 68
CORRIENTES              62
SAN JUAN                61
FORMOSA                 59
CHUBUT                  57
NEUQUEN                 51
SANTA CRUZ              20
TIERRA DEL FUEGO         7
CABA                     1
Name: count, dtype: int64

voy a pasar los datos a PARQUET, de nuevo a la carpeta INTERIM, donde voy a dejar cada archivo que aun falta por procesar

In [193]:
Acc_vel_loc_sinrangos.to_parquet('../../data/interim/Acc_vel_loc_sinrangos.parquet')

---

###
###

---

###
###

# Hoja 2

In [4]:
Accesos_Por_Tecnología = pd.read_csv('../../data/converted/Accesos Por Tecnología.csv')
Accesos_Por_Tecnología.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 986 entries, 0 to 985
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Unnamed: 0    986 non-null    int64  
 1   Año           985 non-null    object 
 2   Trimestre     985 non-null    object 
 3   Provincia     984 non-null    object 
 4   ADSL          984 non-null    float64
 5   Cablemodem    984 non-null    float64
 6   Fibra óptica  984 non-null    float64
 7   Wireless      984 non-null    float64
 8   Otros         984 non-null    float64
 9   Total         984 non-null    float64
dtypes: float64(6), int64(1), object(3)
memory usage: 77.2+ KB


Doy un primer vistazo al DF

In [6]:
Accesos_Por_Tecnología = Accesos_Por_Tecnología.drop('Unnamed: 0', axis=1)
Accesos_Por_Tecnología

Año                                          Trimestre  \
0    2024                                                  1   
1    2024                                                  1   
2    2024                                                  1   
3    2024                                                  1   
4    2024                                                  1   
..    ...                                                ...   
981  2014                                                  1   
982  2014                                                  1   
983  2014                                                  1   
984   NaN                                                NaN   
985     *  Los datos provinciales no coinciden a nivel na...   

               Provincia      ADSL  Cablemodem  Fibra óptica  Wireless  \
0           Buenos Aires  203297.0   2790663.0     1799013.0  150290.0   
1        Capital Federal   70108.0   1177475.0      203756.0    4493.0   
2              Catamarca    5901.0     11025.0       56542.0    1384.0   
3                  Chaco   11343.0     58929.0       64463.0    8404.0   
4                 Chubut   26306.0     77217.0       26356.0   30867.0   
..                   ...       ...         ...           ...       ...   
981  Santiago Del Estero   32567.0      3598.0          19.0     915.0   
982     Tierra Del Fuego   21618.0      2837.0         648.0       1.0   
983              Tucumán  129717.0        83.0         121.0      13.0   
984                  NaN       NaN         NaN           NaN       NaN   
985                  NaN       NaN         NaN           NaN       NaN   

        Otros      Total  
0    101294.0  5044557.0  
1     48879.0  1504711.0  
2        81.0    74933.0  
3     12404.0   155543.0  
4      9930.0   170676.0  
..        ...        ...  
981      34.0    37133.0  
982    2934.0    28038.0  
983      98.0   130032.0  
984       NaN        NaN  
985       NaN        NaN  

[986 rows x 9 columns]

A primera vista se logra encontrar filas nulas.  
Voy a revisar primero la ubicacion de la info(Provincia)

In [7]:
provincia_desconocida = Accesos_Por_Tecnología[Accesos_Por_Tecnología.Provincia.isna()]
provincia_desconocida

Año                                          Trimestre Provincia  ADSL  \
984  NaN                                                NaN       NaN   NaN   
985    *  Los datos provinciales no coinciden a nivel na...       NaN   NaN   

     Cablemodem  Fibra óptica  Wireless  Otros  Total  
984         NaN           NaN       NaN    NaN    NaN  
985         NaN           NaN       NaN    NaN    NaN

Elimino las filas con varios valores nulos

In [8]:
Accesos_Por_Tecnología = Accesos_Por_Tecnología.dropna(subset=['Provincia'])

Ahora es el turno de Trimestre

In [9]:
Accesos_Por_Tecnología.Trimestre.value_counts()

Trimestre
1      240
4      240
3      216
2      216
3 *     24
2 *     24
1 *     24
Name: count, dtype: int64

hay 3 valores mal cargados, vamos a revisarlos

In [10]:
trimestre = Accesos_Por_Tecnología[Accesos_Por_Tecnología.Trimestre=='3 *']
trimestre

Año Trimestre            Provincia       ADSL  Cablemodem  \
432  2019 *       3 *         Buenos Aires  1277249.0   2144956.0   
433  2019 *       3 *      Capital Federal   300288.0   1080973.0   
434  2019 *       3 *            Catamarca    24349.0      5543.0   
435  2019 *       3 *                Chaco    50416.0     49156.0   
436  2019 *       3 *               Chubut    37298.0     27090.0   
437  2019 *       3 *              Córdoba   307780.0    399231.0   
438  2019 *       3 *           Corrientes    53344.0     50230.0   
439  2019 *       3 *           Entre Ríos    94947.0     96305.0   
440  2019 *       3 *              Formosa    21777.0      8359.0   
441  2019 *       3 *                Jujuy    33110.0     35742.0   
442  2019 *       3 *             La Pampa    19469.0     52941.0   
443  2019 *       3 *             La Rioja    14390.0     34984.0   
444  2019 *       3 *              Mendoza   148640.0     54312.0   
445  2019 *       3 *             Misiones    54493.0     65872.0   
446  2019 *       3 *              Neuquén    43347.0     60484.0   
447  2019 *       3 *            Río Negro    67478.0     37425.0   
448  2019 *       3 *                Salta    65799.0     74329.0   
449  2019 *       3 *             San Juan    51670.0      7425.0   
450  2019 *       3 *             San Luis    13037.0     10561.0   
451  2019 *       3 *           Santa Cruz    10663.0     17056.0   
452  2019 *       3 *             Santa Fe   319396.0    377171.0   
453  2019 *       3 *  Santiago Del Estero    24556.0     71198.0   
454  2019 *       3 *     Tierra Del Fuego    19790.0     26659.0   
455  2019 *       3 *              Tucumán   150918.0     12089.0   

     Fibra óptica  Wireless    Otros      Total  
432      425386.0   85572.0  27070.0  3960233.0  
433       24900.0    3895.0  31155.0  1441211.0  
434          13.0     244.0   2251.0    32400.0  
435        8063.0    5224.0   1592.0   114451.0  
436        1917.0   15149.0   7551.0    89005.0  
437       30064.0   37277.0   8701.0   783053.0  
438        1888.0    2458.0   4839.0   112759.0  
439       12888.0   18530.0  15432.0   238102.0  
440        1476.0   14292.0    162.0    46066.0  
441         274.0     812.0  16638.0    86576.0  
442        4994.0   11462.0   1962.0    90828.0  
443          11.0     251.0     41.0    49677.0  
444        6865.0   12975.0   1545.0   224337.0  
445         829.0     419.0  12270.0   133883.0  
446        5560.0   13512.0   2518.0   125421.0  
447        9739.0   10806.0  18847.0   144295.0  
448         733.0    1580.0   9394.0   151835.0  
449        1227.0    7241.0   5535.0    73098.0  
450         350.0   62286.0  36466.0   122700.0  
451         969.0    6544.0   2738.0    37970.0  
452       18690.0   24269.0  21408.0   760934.0  
453         219.0    2759.0  34400.0   133132.0  
454         651.0       0.0   1487.0    48587.0  
455         116.0     684.0    324.0   164131.0

In [11]:
trimestre = Accesos_Por_Tecnología[Accesos_Por_Tecnología.Trimestre=='2 *']
trimestre

Año Trimestre            Provincia       ADSL  Cablemodem  \
456  2019 *       2 *         Buenos Aires  1284658.0   2079900.0   
457  2019 *       2 *      Capital Federal   309937.0   1061358.0   
458  2019 *       2 *            Catamarca    25682.0      5543.0   
459  2019 *       2 *                Chaco    50765.0     47448.0   
460  2019 *       2 *               Chubut    37313.0     26338.0   
461  2019 *       2 *              Córdoba   314768.0    388254.0   
462  2019 *       2 *           Corrientes    53835.0     47431.0   
463  2019 *       2 *           Entre Ríos    97081.0     91081.0   
464  2019 *       2 *              Formosa    21916.0      8686.0   
465  2019 *       2 *                Jujuy    33022.0     30742.0   
466  2019 *       2 *             La Pampa    19471.0     52101.0   
467  2019 *       2 *             La Rioja    15014.0     34984.0   
468  2019 *       2 *              Mendoza   148673.0     54027.0   
469  2019 *       2 *             Misiones    55373.0     64272.0   
470  2019 *       2 *              Neuquén    43420.0     59021.0   
471  2019 *       2 *            Río Negro    67841.0     36527.0   
472  2019 *       2 *                Salta    68827.0     62343.0   
473  2019 *       2 *             San Juan    51680.0      7425.0   
474  2019 *       2 *             San Luis    11130.0     10298.0   
475  2019 *       2 *           Santa Cruz    10665.0     10257.0   
476  2019 *       2 *             Santa Fe   325420.0    355140.0   
477  2019 *       2 *  Santiago Del Estero    24393.0     64783.0   
478  2019 *       2 *     Tierra Del Fuego    19790.0     25527.0   
479  2019 *       2 *              Tucumán   153121.0     12089.0   

     Fibra óptica  Wireless    Otros      Total  
456      382971.0   75240.0  32929.0  3855698.0  
457       36082.0    3962.0   4281.0  1415620.0  
458          13.0     146.0   2251.0    33635.0  
459        4203.0    4727.0   1591.0   108734.0  
460        1917.0   14443.0   7545.0    87556.0  
461       23023.0   33847.0  15236.0   775128.0  
462        1701.0    2444.0   6138.0   111549.0  
463       12764.0   17369.0  17529.0   235824.0  
464        1382.0   14292.0    162.0    46438.0  
465         274.0     812.0  21857.0    86707.0  
466        4840.0   10789.0   3004.0    90205.0  
467          11.0     251.0     41.0    50301.0  
468        6865.0   12748.0   1544.0   223857.0  
469         543.0     421.0  12269.0   132878.0  
470        4983.0   12792.0   2737.0   122953.0  
471        4037.0   10485.0  19657.0   138547.0  
472         725.0    1210.0   9463.0   142568.0  
473        1174.0    2870.0   6876.0    70025.0  
474         347.0   57242.0  38368.0   117385.0  
475         379.0    6514.0   2735.0    30550.0  
476       18236.0   21980.0  22708.0   743484.0  
477         219.0    2759.0  34600.0   126754.0  
478         651.0       0.0   1486.0    47454.0  
479         116.0     684.0    321.0   166331.0

In [12]:
trimestre = Accesos_Por_Tecnología[Accesos_Por_Tecnología.Trimestre=='1 *']
trimestre

Año Trimestre            Provincia       ADSL  Cablemodem  \
480  2019 *       1 *         Buenos Aires  1304507.0   2020730.0   
481  2019 *       1 *      Capital Federal   328246.0   1030814.0   
482  2019 *       1 *            Catamarca    26912.0      4443.0   
483  2019 *       1 *                Chaco    53409.0     43555.0   
484  2019 *       1 *               Chubut    37302.0     23065.0   
485  2019 *       1 *              Córdoba   323272.0    363959.0   
486  2019 *       1 *           Corrientes    55842.0     43444.0   
487  2019 *       1 *           Entre Ríos   100999.0     90253.0   
488  2019 *       1 *              Formosa    22570.0      7878.0   
489  2019 *       1 *                Jujuy    33248.0     12484.0   
490  2019 *       1 *             La Pampa    18832.0     57373.0   
491  2019 *       1 *             La Rioja    16367.0     34984.0   
492  2019 *       1 *              Mendoza   148524.0     40205.0   
493  2019 *       1 *             Misiones    54898.0     59214.0   
494  2019 *       1 *              Neuquén    40747.0     55634.0   
495  2019 *       1 *            Río Negro    68377.0     34124.0   
496  2019 *       1 *                Salta    74426.0     57933.0   
497  2019 *       1 *             San Juan    51660.0      5838.0   
498  2019 *       1 *             San Luis    11108.0      7784.0   
499  2019 *       1 *           Santa Cruz    10659.0      4615.0   
500  2019 *       1 *             Santa Fe   340038.0    340344.0   
501  2019 *       1 *  Santiago Del Estero    25337.0     61547.0   
502  2019 *       1 *     Tierra Del Fuego    19789.0     23549.0   
503  2019 *       1 *              Tucumán   159215.0     11670.0   

     Fibra óptica  Wireless    Otros      Total  
480      336362.0   64867.0  20201.0  3746667.0  
481       32281.0    3978.0   4281.0  1399600.0  
482          16.0       0.0   2251.0    33622.0  
483        1951.0    3628.0   1134.0   103677.0  
484        1923.0   14443.0   3045.0    79778.0  
485       19902.0   24615.0  12367.0   744115.0  
486        1615.0    1528.0     58.0   102487.0  
487       12723.0   16231.0  12241.0   232447.0  
488         940.0   13761.0     62.0    45211.0  
489         153.0     640.0  43622.0    90147.0  
490        2891.0    9023.0   1041.0    89160.0  
491          12.0     252.0     41.0    51656.0  
492        5662.0   10363.0   1544.0   206298.0  
493         465.0     426.0   7717.0   122720.0  
494        3114.0   12510.0   2372.0   114377.0  
495        3994.0    9736.0   4857.0   121088.0  
496         647.0    1135.0   1879.0   136020.0  
497         808.0    2932.0   1207.0    62445.0  
498         378.0   53097.0  23968.0    96335.0  
499         382.0    6514.0   2735.0    24905.0  
500       15621.0   23215.0  17860.0   737078.0  
501          23.0    2847.0     99.0    89853.0  
502        1623.0       0.0   1486.0    46447.0  
503         118.0     678.0    321.0   172002.0

Los 3 trimestres corresponden al año 2019, el cual tambien tiene un asterisco (que vamos a revisar luego), antes de modificarlo voy a revisar el año mismo para chequear si tiene datos faltantes

In [13]:
anio = Accesos_Por_Tecnología[Accesos_Por_Tecnología['Año']=='2019']

In [14]:
anio

Año Trimestre            Provincia      ADSL  Cablemodem  Fibra óptica  \
408  2019         4         Buenos Aires  720011.0   2191672.0      669194.0   
409  2019         4      Capital Federal  196938.0   1093143.0       35534.0   
410  2019         4            Catamarca   22999.0      3318.0        9506.0   
411  2019         4                Chaco   48661.0     53602.0        8831.0   
412  2019         4               Chubut   47045.0     26977.0        1324.0   
413  2019         4              Córdoba  296872.0    410038.0       53335.0   
414  2019         4           Corrientes   52525.0     51845.0        2002.0   
415  2019         4           Entre Ríos   92981.0     99206.0       12875.0   
416  2019         4              Formosa   21567.0      8684.0        1760.0   
417  2019         4                Jujuy   32949.0     37604.0         873.0   
418  2019         4             La Pampa   15457.0     56854.0        4942.0   
419  2019         4             La Rioja   13738.0     34984.0           0.0   
420  2019         4              Mendoza   96678.0     49711.0       24989.0   
421  2019         4             Misiones   53116.0     69676.0         847.0   
422  2019         4              Neuquén   29826.0     60970.0       10549.0   
423  2019         4            Río Negro   49747.0     37933.0        9658.0   
424  2019         4                Salta   63272.0     76062.0       21865.0   
425  2019         4             San Juan   41360.0      6338.0        4931.0   
426  2019         4             San Luis    8569.0      8027.0           6.0   
427  2019         4           Santa Cruz   13264.0     15899.0        1036.0   
428  2019         4             Santa Fe  312072.0    382254.0       27464.0   
429  2019         4  Santiago Del Estero   24682.0     71239.0         200.0   
430  2019         4     Tierra Del Fuego   13261.0     26821.0           0.0   
431  2019         4              Tucumán  146985.0     11012.0       39574.0   

     Wireless    Otros      Total  
408   87111.0  18266.0  3686254.0  
409    3914.0  28576.0  1358105.0  
410     261.0   2227.0    38311.0  
411    5229.0   1580.0   117903.0  
412   15267.0   5489.0    96102.0  
413   37750.0   8693.0   806688.0  
414    2458.0   5004.0   113834.0  
415   18561.0  15407.0   239030.0  
416   14531.0    169.0    46711.0  
417     814.0  16641.0    88881.0  
418   11341.0   1805.0    90399.0  
419     251.0     29.0    49002.0  
420   13672.0    730.0   185780.0  
421     441.0  11733.0   135813.0  
422   14246.0    854.0   116445.0  
423   11437.0  18365.0   127140.0  
424    3040.0   8206.0   172445.0  
425    7241.0   5104.0    64974.0  
426   55266.0  36333.0   108201.0  
427    6800.0   2371.0    39370.0  
428   26978.0  20401.0   769169.0  
429    2810.0   4657.0   103588.0  
430       0.0    310.0    40392.0  
431     725.0    348.0   198644.0

Habiendo chequeado que faltan filas voy a corregir el error en los trimestres, reemplazando en el string de trimestre el '1 *' por un '1'

In [15]:
Accesos_Por_Tecnología.Trimestre = Accesos_Por_Tecnología.Trimestre.str.replace(' *','') 

In [16]:
Accesos_Por_Tecnología.Trimestre.value_counts()

Trimestre
1    264
4    240
3    240
2    240
Name: count, dtype: int64

Finalmente revisamos la columna Año

In [17]:
Accesos_Por_Tecnología.Año.value_counts()

Año
2023      96
2022      96
2021      96
2020      96
2015      96
2018      96
2016      96
2017      96
2014      96
2019 *    72
2024      24
2019      24
Name: count, dtype: int64

Reemplazamos el valor mal cargado

In [18]:
Accesos_Por_Tecnología.Año = Accesos_Por_Tecnología.Año.str.replace(' *','')

In [19]:
Accesos_Por_Tecnología.Año.value_counts()

Año
2023    96
2022    96
2021    96
2017    96
2020    96
2019    96
2018    96
2015    96
2016    96
2014    96
2024    24
Name: count, dtype: int64

In [20]:
Accesos_Por_Tecnología.to_parquet('../../data/interim/accesos_tecnologia')

---



###
###

---

###
###

# Hoja 3

In [210]:
Accesos_por_velocidad = pd.read_csv('../../data/converted/Accesos por velocidad.csv')
Accesos_por_velocidad

Unnamed: 0   Año  Trimestre            Provincia  HASTA 512 kbps  \
0             0  2024          1         Buenos Aires         26002.0   
1             1  2024          1      Capital Federal           516.0   
2             2  2024          1            Catamarca            71.0   
3             3  2024          1                Chaco           236.0   
4             4  2024          1               Chubut           111.0   
..          ...   ...        ...                  ...             ...   
979         979  2014          1           Santa Cruz           161.0   
980         980  2014          1             Santa Fe          8456.0   
981         981  2014          1  Santiago Del Estero          1234.0   
982         982  2014          1     Tierra Del Fuego            12.0   
983         983  2014          1              Tucumán             6.0   

     + 512 Kbps - 1 Mbps  + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  \
0                22510.0           221185.0            233225.0   
1                 5418.0            29753.0             51415.0   
2                  176.0             2872.0              3424.0   
3                  416.0            14156.0              9044.0   
4                 1083.0            43828.0             31885.0   
..                   ...                ...                 ...   
979               1625.0            24972.0                 1.0   
980             124468.0           345225.0             20328.0   
981              10531.0            22817.0              2422.0   
982                607.0            30902.0                 6.0   
983              34672.0            83210.0             11779.0   

     + 10 Mbps - 20 Mbps  + 20 Mbps - 30 Mbps  + 30 Mbps     OTROS      Total  
0               241778.0             106418.0  4068292.0  125147.0  5044557.0  
1                37369.0               5957.0  1374283.0       0.0  1504711.0  
2                 4204.0               1822.0    62294.0      70.0    74933.0  
3                 5533.0               5718.0   116624.0    3816.0   155543.0  
4                23024.0              20261.0    33844.0   16640.0   170676.0  
..                   ...                  ...        ...       ...        ...  
979                  1.0                  0.0        0.0       0.0    26760.0  
980               6845.0                 23.0      668.0       0.0   506013.0  
981                109.0                  0.0        0.0       0.0    37113.0  
982                  0.0                  0.0        0.0       0.0    31527.0  
983                362.0                  3.0        0.0       0.0   130032.0  

[984 rows x 13 columns]

In [211]:
Accesos_por_velocidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Unnamed: 0           984 non-null    int64  
 1   Año                  984 non-null    int64  
 2   Trimestre            984 non-null    int64  
 3   Provincia            984 non-null    object 
 4   HASTA 512 kbps       984 non-null    float64
 5   + 512 Kbps - 1 Mbps  984 non-null    float64
 6   + 1 Mbps - 6 Mbps    984 non-null    float64
 7   + 6 Mbps - 10 Mbps   984 non-null    float64
 8   + 10 Mbps - 20 Mbps  984 non-null    float64
 9   + 20 Mbps - 30 Mbps  984 non-null    float64
 10  + 30 Mbps            984 non-null    float64
 11  OTROS                978 non-null    float64
 12  Total                984 non-null    float64
dtypes: float64(9), int64(3), object(1)
memory usage: 100.1+ KB


In [212]:
otros_nulos = Accesos_por_velocidad[Accesos_por_velocidad.OTROS.isna()]
otros_nulos

Unnamed: 0   Año  Trimestre         Provincia  HASTA 512 kbps  \
457         457  2019          2   Capital Federal            74.0   
467         467  2019          2          La Rioja             9.0   
468         468  2019          2           Mendoza            20.0   
475         475  2019          2        Santa Cruz            25.0   
478         478  2019          2  Tierra Del Fuego             6.0   
479         479  2019          2           Tucumán            16.0   

     + 512 Kbps - 1 Mbps  + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  \
457               1027.0           281370.0            124087.0   
467                 11.0            39819.0              4297.0   
468               2522.0           205165.0              2971.0   
475                 84.0            24861.0              3806.0   
478                123.0            41824.0              4241.0   
479                157.0            78891.0             44296.0   

     + 10 Mbps - 20 Mbps  + 20 Mbps - 30 Mbps  + 30 Mbps  OTROS      Total  
457             239691.0             296155.0   475991.0    NaN  1418395.0  
467               4857.0                 13.0     1298.0    NaN    50304.0  
468              10092.0               1256.0      234.0    NaN   222260.0  
475               1202.0                 28.0        4.0    NaN    30010.0  
478               1573.0                 96.0       32.0    NaN    47895.0  
479              24291.0                628.0    18394.0    NaN   166673.0

In [213]:
Accesos_por_velocidad = Accesos_por_velocidad.fillna(0)

In [214]:
anio2023 = Accesos_por_velocidad[Accesos_por_velocidad.Año == 2023]
anio2023

Unnamed: 0   Año  Trimestre            Provincia  HASTA 512 kbps  \
48           48  2023          3         Buenos Aires         27440.0   
49           49  2023          3      Capital Federal           516.0   
50           50  2023          3            Catamarca            71.0   
51           51  2023          3                Chaco           236.0   
52           52  2023          3               Chubut           111.0   
..          ...   ...        ...                  ...             ...   
115         115  2023          1           Santa Cruz            51.0   
116         116  2023          1             Santa Fe           468.0   
117         117  2023          1  Santiago Del Estero           178.0   
118         118  2023          1     Tierra Del Fuego            22.0   
119         119  2023          1              Tucumán            39.0   

     + 512 Kbps - 1 Mbps  + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  \
48               24820.0           242010.0            241310.0   
49                4690.0            28089.0             50899.0   
50                 224.0             3194.0              3893.0   
51                 462.0            14076.0             12874.0   
52                1470.0            44434.0             30908.0   
..                   ...                ...                 ...   
115               6810.0            11528.0             11380.0   
116               7655.0            98198.0            180708.0   
117               1186.0            21551.0             57189.0   
118                831.0            10944.0             13048.0   
119               1237.0            23379.0             29629.0   

     + 10 Mbps - 20 Mbps  + 20 Mbps - 30 Mbps  + 30 Mbps     OTROS      Total  
48              246076.0             112159.0  3951660.0  122762.0  4968237.0  
49               34061.0               6020.0  1343709.0       0.0  1467984.0  
50                4634.0               2124.0    60123.0      92.0    74355.0  
51                6895.0               2970.0   100544.0    4089.0   142146.0  
52               24394.0              19608.0    30565.0   16578.0   168068.0  
..                   ...                  ...        ...       ...        ...  
115               9861.0               3697.0     5990.0    1044.0    50361.0  
116              75061.0              26306.0   479777.0   18101.0   886274.0  
117              21277.0                734.0    15447.0     993.0   118555.0  
118              21348.0               3959.0     2386.0    4575.0    57113.0  
119              25565.0               1723.0   164402.0   15126.0   261100.0  

[72 rows x 13 columns]

In [215]:
trimestre4 = Accesos_por_velocidad[Accesos_por_velocidad.Trimestre==4]
trimestre4

Unnamed: 0   Año  Trimestre            Provincia  HASTA 512 kbps  \
24           24  2024          4         Buenos Aires         26909.0   
25           25  2024          4      Capital Federal           516.0   
26           26  2024          4            Catamarca            71.0   
27           27  2024          4                Chaco           236.0   
28           28  2024          4               Chubut           111.0   
..          ...   ...        ...                  ...             ...   
907         907  2014          4           Santa Cruz           156.0   
908         908  2014          4             Santa Fe          8707.0   
909         909  2014          4  Santiago Del Estero          1246.0   
910         910  2014          4     Tierra Del Fuego            12.0   
911         911  2014          4              Tucumán            17.0   

     + 512 Kbps - 1 Mbps  + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  \
24               23962.0           228393.0            239339.0   
25                4544.0            26409.0             48083.0   
26                 209.0             3065.0              3734.0   
27                 477.0            14811.0             10361.0   
28                1088.0            44276.0             31402.0   
..                   ...                ...                 ...   
907               1400.0            24145.0                 1.0   
908              99962.0           359293.0             36718.0   
909               8913.0            24181.0              4359.0   
910                283.0            31489.0                 2.0   
911              28746.0            85091.0             20631.0   

     + 10 Mbps - 20 Mbps  + 20 Mbps - 30 Mbps  + 30 Mbps    OTROS      Total  
24              246144.0             110876.0  4042646.0  93351.0  5011620.0  
25               33925.0               5829.0  1355061.0      0.0  1474367.0  
26                4423.0               1827.0    61181.0     77.0    74587.0  
27                6247.0               5729.0   105485.0   4040.0   147386.0  
28               22843.0              20196.0    33665.0  16589.0   170170.0  
..                   ...                  ...        ...      ...        ...  
907                  1.0                  0.0        0.0      0.0    25703.0  
908              21098.0                869.0      824.0      0.0   527471.0  
909                602.0                  0.0        0.0      0.0    39301.0  
910                  0.0                  0.0        0.0      0.0    31786.0  
911               2733.0                196.0        0.0      0.0   137414.0  

[240 rows x 13 columns]

In [216]:
trimestre4.info()

<class 'pandas.core.frame.DataFrame'>
Index: 240 entries, 24 to 911
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Unnamed: 0           240 non-null    int64  
 1   Año                  240 non-null    int64  
 2   Trimestre            240 non-null    int64  
 3   Provincia            240 non-null    object 
 4   HASTA 512 kbps       240 non-null    float64
 5   + 512 Kbps - 1 Mbps  240 non-null    float64
 6   + 1 Mbps - 6 Mbps    240 non-null    float64
 7   + 6 Mbps - 10 Mbps   240 non-null    float64
 8   + 10 Mbps - 20 Mbps  240 non-null    float64
 9   + 20 Mbps - 30 Mbps  240 non-null    float64
 10  + 30 Mbps            240 non-null    float64
 11  OTROS                240 non-null    float64
 12  Total                240 non-null    float64
dtypes: float64(9), int64(3), object(1)
memory usage: 26.2+ KB


In [217]:
trimestre4.Año.value_counts()

Año
2024    24
2022    24
2021    24
2020    24
2019    24
2018    24
2017    24
2016    24
2015    24
2014    24
Name: count, dtype: int64

Me falta el dato del 4 trimestre del año 2023, por el momento, no lo voy a sumar a los DF. lo voy a dejar por el momento fuera del analisis del EDA  
Corroboro que la totalidad de los años coincidad con la cantidad de datos

In [218]:
Accesos_por_velocidad.Año.value_counts().sum()

np.int64(984)

---


###
###

---

###
###

# Hoja 4

In [219]:
Accesos_tecnologia_localidad = pd.read_csv("../../data/converted/Accesos_tecnologia_localidad.csv")
Accesos_tecnologia_localidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3352 entries, 0 to 3351
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     3352 non-null   int64 
 1   Provincia      3072 non-null   object
 2   Partido        3072 non-null   object
 3   Localidad      3072 non-null   object
 4   Link Indec     3072 non-null   object
 5   ADSL           3352 non-null   int64 
 6   CABLEMODEM     3352 non-null   int64 
 7   DIAL UP        3352 non-null   int64 
 8   FIBRA OPTICA   3352 non-null   int64 
 9   OTROS          3352 non-null   int64 
 10  SATELITAL      3352 non-null   int64 
 11  WIMAX          3352 non-null   int64 
 12  WIRELESS       3352 non-null   int64 
 13  Total general  3352 non-null   int64 
dtypes: int64(10), object(4)
memory usage: 366.8+ KB


Con info ya puedo ver que faltan datos y que los 3 datos que me dan la ubicacion coinciden, por lo que voy a usar dropna(subset= 'provincia'/'partido'/'localidad')

In [220]:
Accesos_tecnologia_localidad = Accesos_tecnologia_localidad.dropna(subset='Provincia')
Accesos_tecnologia_localidad.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3072 entries, 0 to 3071
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     3072 non-null   int64 
 1   Provincia      3072 non-null   object
 2   Partido        3072 non-null   object
 3   Localidad      3072 non-null   object
 4   Link Indec     3072 non-null   object
 5   ADSL           3072 non-null   int64 
 6   CABLEMODEM     3072 non-null   int64 
 7   DIAL UP        3072 non-null   int64 
 8   FIBRA OPTICA   3072 non-null   int64 
 9   OTROS          3072 non-null   int64 
 10  SATELITAL      3072 non-null   int64 
 11  WIMAX          3072 non-null   int64 
 12  WIRELESS       3072 non-null   int64 
 13  Total general  3072 non-null   int64 
dtypes: int64(10), object(4)
memory usage: 360.0+ KB


In [221]:
Accesos_tecnologia_localidad

Unnamed: 0     Provincia      Partido  \
0              0  BUENOS AIRES   25 de Mayo   
1              1  BUENOS AIRES   25 de Mayo   
2              2  BUENOS AIRES   25 de Mayo   
3              3  BUENOS AIRES   25 de Mayo   
4              4  BUENOS AIRES   25 de Mayo   
...          ...           ...          ...   
3067        3067       TUCUMAN   Tafí Viejo   
3068        3068       TUCUMAN      Trancas   
3069        3069       TUCUMAN      Trancas   
3070        3070       TUCUMAN  Yerba Buena   
3071        3071       TUCUMAN  Yerba Buena   

                               Localidad Link Indec  ADSL  CABLEMODEM  \
0                             25 de Mayo    6854100   755        4600   
1                              Del Valle    6854020   181           0   
2                      Gobernador Ugarte    6854040     0           0   
3                 Norberto de la Riestra    6854060     0         782   
4                       Lucas Monteverde    6854050     0           0   
...                                  ...        ...   ...         ...   
3067  Villa Mariano Moreno - El Colmenar   90105100     0           0   
3068                San Pedro de Colalao   90112020     0           0   
3069                   Villa  de Trancas   90112030    95           0   
3070                       Villa Carmela   90119020   636           0   
3071            Yerba Buena - Marcos Paz   90119030   129          46   

      DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  WIRELESS  Total general  
0           0             2      0        742      0       727           6826  
1           0             0     10          1      0         0            192  
2           0             0      0          0      0       181            181  
3           0           496     16          0      0       185           1479  
4           0             0      0          0      0         6              6  
...       ...           ...    ...        ...    ...       ...            ...  
3067        0          7422      0          0      0         0           7422  
3068        0             0      0          0      0       424            424  
3069        0             0     33          0      0        71            199  
3070        0             0      1          0      0        79            716  
3071        0         17796     33          0      0         1          18005  

[3072 rows x 14 columns]

In [222]:
Accesos_tecnologia_localidad.Provincia.value_counts()

Provincia
BUENOS AIRES           715
CORDOBA                388
SANTA FE               355
ENTRE RIOS             141
SANTIAGO DEL ESTERO    137
MENDOZA                133
SALTA                  122
San Luis                95
RIO NEGRO               93
MISIONES                93
JUJUY                   91
LA PAMPA                83
CHACO                   76
LA RIOJA                75
CATAMARCA               74
TUCUMAN                 70
SAN JUAN                67
CORRIENTES              63
FORMOSA                 62
CHUBUT                  59
NEUQUEN                 52
SANTA CRUZ              20
TIERRA DEL FUEGO         7
CABA                     1
Name: count, dtype: int64

---

###
###

---

###
###

# Hoja 5

In [223]:
Dial_BAf = pd.read_csv("../../data/converted/Dial-BAf.csv")
Dial_BAf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        984 non-null    int64  
 1   Año               984 non-null    int64  
 2   Trimestre         984 non-null    int64  
 3   Provincia         984 non-null    object 
 4   Banda ancha fija  984 non-null    int64  
 5   Dial up           982 non-null    float64
 6   Total             984 non-null    int64  
dtypes: float64(1), int64(5), object(1)
memory usage: 53.9+ KB


In [224]:
Dial_BAf

Unnamed: 0   Año  Trimestre            Provincia  Banda ancha fija  \
0             0  2024          1         Buenos Aires           5038657   
1             1  2024          1      Capital Federal           1502661   
2             2  2024          1            Catamarca             74932   
3             3  2024          1                Chaco            155538   
4             4  2024          1               Chubut            169819   
..          ...   ...        ...                  ...               ...   
979         979  2014          1           Santa Cruz             26304   
980         980  2014          1             Santa Fe            506000   
981         981  2014          1  Santiago Del Estero             37124   
982         982  2014          1     Tierra Del Fuego             27272   
983         983  2014          1              Tucumán            129944   

     Dial up    Total  
0     5900.0  5044557  
1     2050.0  1504711  
2        1.0    74933  
3        5.0   155543  
4      857.0   170676  
..       ...      ...  
979    460.0    26764  
980    612.0   506612  
981      9.0    37133  
982    766.0    28038  
983     88.0   130032  

[984 rows x 7 columns]

In [225]:
print(Dial_BAf.Año.value_counts())
print(Dial_BAf.Trimestre.value_counts())
print(Dial_BAf.Provincia.value_counts())

Año
2023    96
2022    96
2021    96
2017    96
2020    96
2019    96
2018    96
2015    96
2016    96
2014    96
2024    24
Name: count, dtype: int64
Trimestre
1    264
4    240
3    240
2    240
Name: count, dtype: int64
Provincia
Buenos Aires           41
Capital Federal        41
Catamarca              41
Chaco                  41
Chubut                 41
Córdoba                41
Corrientes             41
Entre Ríos             41
Formosa                41
Jujuy                  41
La Pampa               41
La Rioja               41
Mendoza                41
Misiones               41
Neuquén                41
Río Negro              41
Salta                  41
San Juan               41
San Luis               41
Santa Cruz             41
Santa Fe               41
Santiago Del Estero    41
Tierra Del Fuego       41
Tucumán                41
Name: count, dtype: int64


No veo nada raro, se deja tal cual 

---

###

---

###

# Hoja 6

In [9]:
Ingresos = pd.read_csv("../../data/converted/Ingresos .csv")
Ingresos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 5 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Unnamed: 0                 41 non-null     int64  
 1   Año                        41 non-null     int64  
 2   Trimestre                  41 non-null     int64  
 3   Ingresos (miles de pesos)  41 non-null     float64
 4   Periodo                    41 non-null     object 
dtypes: float64(1), int64(3), object(1)
memory usage: 1.7+ KB


**Encontramos un año mal ingresado

In [10]:
Ingresos.head()

Unnamed: 0   Año  Trimestre  Ingresos (miles de pesos)        Periodo
0           0  2024          1               2.804191e+08   Ene-Mar 2024
1           1  2023          4               1.673760e+08   Oct-Dic 2023
2           2  2023          3               1.331066e+08  Jul-Sept 2023
3           3  2023          2               1.180603e+08   Jun-Mar 2023
4           4  2033          1               1.041350e+08   Ene-Mar 2023

In [11]:
Ingresos.Año.value_counts()

Año
2022    4
2019    4
2020    4
2021    4
2017    4
2016    4
2015    4
2018    4
2014    4
2023    3
2033    1
2024    1
Name: count, dtype: int64

In [12]:
Ingresos.Año = Ingresos.Año.replace(2033, 2023)
Ingresos.Año.value_counts()

Año
2023    4
2022    4
2021    4
2017    4
2020    4
2019    4
2018    4
2015    4
2016    4
2014    4
2024    1
Name: count, dtype: int64

In [13]:
Ingresos = Ingresos.drop('Unnamed: 0',axis=1)

In [14]:
Ingresos.to_parquet("../../data/interim/ingresos.parquet")

---

###
###

---

###
###

# Hoja 7

In [230]:
Penetracion_hogares = pd.read_csv("../../data/converted/Penetracion-hogares.csv")
Penetracion_hogares.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Unnamed: 0                    984 non-null    int64  
 1   Año                           984 non-null    int64  
 2   Trimestre                     984 non-null    int64  
 3   Provincia                     984 non-null    object 
 4   Accesos por cada 100 hogares  984 non-null    float64
dtypes: float64(1), int64(3), object(1)
memory usage: 38.6+ KB


In [231]:
Penetracion_hogares

Unnamed: 0   Año  Trimestre            Provincia  \
0             0  2024          1         Buenos Aires   
1             1  2024          1      Capital Federal   
2             2  2024          1            Catamarca   
3             3  2024          1                Chaco   
4             4  2024          1               Chubut   
..          ...   ...        ...                  ...   
979         979  2014          1           Santa Cruz   
980         980  2014          1             Santa Fe   
981         981  2014          1  Santiago Del Estero   
982         982  2014          1     Tierra Del Fuego   
983         983  2014          1              Tucumán   

     Accesos por cada 100 hogares  
0                       81.100000  
1                      119.530000  
2                       68.820000  
3                       46.700000  
4                       84.890000  
..                            ...  
979                     28.939093  
980                     47.847663  
981                     16.309924  
982                     63.965926  
983                     33.420975  

[984 rows x 5 columns]

In [232]:
Penetracion_hogares.Provincia.value_counts()

Provincia
Buenos Aires           41
Capital Federal        41
Catamarca              41
Chaco                  41
Chubut                 41
Córdoba                41
Corrientes             41
Entre Ríos             41
Formosa                41
Jujuy                  41
La Pampa               41
La Rioja               41
Mendoza                41
Misiones               41
Neuquén                41
Río Negro              41
Salta                  41
San Juan               41
San Luis               41
Santa Cruz             41
Santa Fe               41
Santiago Del Estero    41
Tierra Del Fuego       41
Tucumán                41
Name: count, dtype: int64

---

###
###

---

###
###

# Hoja 8

In [15]:
Penetracion_totales = pd.read_csv("../../data/converted/Penetracion-totales.csv")
Penetracion_totales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 6 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Unnamed: 0                    41 non-null     int64  
 1   Año                           41 non-null     int64  
 2   Trimestre                     41 non-null     int64  
 3   Accesos por cada 100 hogares  41 non-null     float64
 4   Accesos por cada 100 hab      41 non-null     float64
 5   Periodo                       41 non-null     object 
dtypes: float64(2), int64(3), object(1)
memory usage: 2.0+ KB


In [16]:
Penetracion_totales

Unnamed: 0   Año  Trimestre  Accesos por cada 100 hogares  \
0            0  2024          1                     78.890000   
1            1  2023          4                     78.560000   
2            2  2023          3                     77.840000   
3            3  2023          2                     77.020000   
4            4  2023          1                     77.200000   
5            5  2022          4                     77.210000   
6            6  2022          3                     76.640000   
7            7  2022          2                     75.965424   
8            8  2022          1                     73.878803   
9            9  2021          4                     73.176397   
10          10  2021          3                     70.578853   
11          11  2021          2                     69.240183   
12          12  2021          1                     67.945876   
13          13  2020          4                     67.622339   
14          14  2020          3                     66.310000   
15          15  2020          2                     64.210003   
16          16  2020          1                     62.856361   
17          17  2019          4                     62.919912   
18          18  2019          3                     65.787985   
19          19  2019          2                     64.530000   
20          20  2019          1                     62.479166   
21          21  2018          4                     61.415807   
22          22  2018          3                     60.690000   
23          23  2018          2                     59.162069   
24          24  2018          1                     58.820000   
25          25  2017          4                     57.782351   
26          26  2017          3                     56.899221   
27          27  2017          2                     54.901191   
28          28  2017          1                     54.162116   
29          29  2016          4                     53.936019   
30          30  2016          3                     53.734186   
31          31  2016          2                     53.342201   
32          32  2016          1                     51.845452   
33          33  2015          4                     52.630888   
34          34  2015          3                     52.459106   
35          35  2015          2                     51.760266   
36          36  2015          1                     51.364586   
37          37  2014          4                     50.499788   
38          38  2014          3                     50.672368   
39          39  2014          2                     49.861428   
40          40  2014          1                     49.545821   

    Accesos por cada 100 hab        Periodo  
0                  24.790000   Ene-Mar 2024  
1                  24.660000   Oct-Dic 2023  
2                  24.410000  Jul-Sept 2023  
3                  24.140000   Abr-Jun 2023  
4                  24.170000   Ene-Mar 2023  
5                  24.150000   Oct-Dic 2022  
6                  23.950000  Jul-Sept 2022  
7                  23.720442   Abr-Jun 2022  
8                  23.049171   Ene-Mar 2022  
9                  22.810702   Oct-Dic 2021  
10                 21.982550  Jul-Sept 2021  
11                 21.547939   Abr-Jun 2021  
12                 21.127832   Ene-Mar 2021  
13                 21.010192   Oct-Dic 2020  
14                 20.590000  Jul-Sept 2020  
15                 19.918403   Abr-Jun 2020  
16                 19.483265   Ene-Mar 2020  
17                 19.487900   Oct-Dic 2019  
18                 20.360661  Jul-Sept 2019  
19                 19.960000   Abr-Jun 2019  
20                 19.307907   Ene-Mar 2019  
21                 18.970000   Oct-Dic 2018  
22                 18.730000  Jul-Sept 2018  
23                 18.240000   Abr-Jun 2018  
24                 18.120000   Ene-Mar 2018  
25                 17.792977   Oct-Dic 2017  
26                 17.509032  J

In [17]:
Penetracion_totales.Periodo.value_counts()

Periodo
Ene-Mar 2024     1
Oct-Dic 2023     1
Jul-Sept 2023    1
Abr-Jun 2023     1
Ene-Mar 2023     1
Oct-Dic 2022     1
Jul-Sept 2022    1
Abr-Jun 2022     1
Ene-Mar 2022     1
Oct-Dic 2021     1
Jul-Sept 2021    1
Abr-Jun 2021     1
Ene-Mar 2021     1
Oct-Dic 2020     1
Jul-Sept 2020    1
Abr-Jun 2020     1
Ene-Mar 2020     1
Oct-Dic 2019     1
Jul-Sept 2019    1
Abr-Jun 2019     1
Ene-Mar 2019     1
Oct-Dic 2018     1
Jul-Sept 2018    1
Abr-Jun 2018     1
Ene-Mar 2018     1
Oct-Dic 2017     1
Jul-Sept 2017    1
Abr-Jun 2017     1
Ene-Mar 2017     1
Oct-Dic 2016     1
Jul-Sept 2016    1
Abr-Jun 2016     1
Ene-Mar 2016     1
Oct-Dic 2015     1
Jul-Sept 2015    1
Abr-Jun 2015     1
Ene-Mar 2015     1
Oct-Dic 2014     1
Jul-Sept 2014    1
Abr-Jun 2014     1
Ene-Mar 2014     1
Name: count, dtype: int64

In [18]:
Penetracion_totales.to_parquet("../../data/interim/penetracion_totales.parquet")

---

###
###

---

###
###

# Hoja 9

In [236]:
Penetración_poblacion = pd.read_csv("../../data/converted/Penetración-poblacion.csv")
Penetración_poblacion.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                984 non-null    int64  
 1   Año                       984 non-null    int64  
 2   Trimestre                 984 non-null    int64  
 3   Provincia                 984 non-null    object 
 4   Accesos por cada 100 hab  984 non-null    float64
dtypes: float64(1), int64(3), object(1)
memory usage: 38.6+ KB


In [237]:
Penetración_poblacion

Unnamed: 0   Año  Trimestre            Provincia  \
0             0  2024          1         Buenos Aires   
1             1  2024          1      Capital Federal   
2             2  2024          1            Catamarca   
3             3  2024          1                Chaco   
4             4  2024          1               Chubut   
..          ...   ...        ...                  ...   
979         979  2014          1           Santa Cruz   
980         980  2014          1             Santa Fe   
981         981  2014          1  Santiago Del Estero   
982         982  2014          1     Tierra Del Fuego   
983         983  2014          1              Tucumán   

     Accesos por cada 100 hab  
0                   27.800000  
1                   48.780000  
2                   17.500000  
3                   12.480000  
4                   26.020000  
..                        ...  
979                  8.677327  
980                 15.077801  
981                  4.058899  
982                 19.104921  
983                  8.307215  

[984 rows x 5 columns]

In [238]:
Penetración_poblacion.Provincia.value_counts()

Provincia
Buenos Aires           41
Capital Federal        41
Catamarca              41
Chaco                  41
Chubut                 41
Córdoba                41
Corrientes             41
Entre Ríos             41
Formosa                41
Jujuy                  41
La Pampa               41
La Rioja               41
Mendoza                41
Misiones               41
Neuquén                41
Río Negro              41
Salta                  41
San Juan               41
San Luis               41
Santa Cruz             41
Santa Fe               41
Santiago Del Estero    41
Tierra Del Fuego       41
Tucumán                41
Name: count, dtype: int64

---

###
###

---

###
###

# Hoja 10

In [239]:
Totales_Accesos_Por_Tecnología = pd.read_csv("../../data/converted/Totales Accesos Por Tecnología.csv")
Totales_Accesos_Por_Tecnología.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Unnamed: 0    41 non-null     int64 
 1   Año           41 non-null     int64 
 2   Trimestre     41 non-null     int64 
 3   ADSL          41 non-null     int64 
 4   Cablemodem    41 non-null     int64 
 5   Fibra óptica  41 non-null     int64 
 6   Wireless      41 non-null     int64 
 7   Otros         41 non-null     int64 
 8   Total         41 non-null     int64 
 9   Periodo       41 non-null     object
dtypes: int64(9), object(1)
memory usage: 3.3+ KB


In [240]:
Totales_Accesos_Por_Tecnología

Unnamed: 0   Año  Trimestre     ADSL  Cablemodem  Fibra óptica  Wireless  \
0            0  2024          1   774475     5986957       4015101    598682   
1            1  2023          4   836390     6022532       3908183    585760   
2            2  2023          3   897895     6018832       3708718    581436   
3            3  2023          2  1006509     5997149       3463988    581823   
4            4  2023          1  1204417     6002582       3243411    576647   
5            5  2022          4  1315373     6037457       3060575    572969   
6            6  2022          3  1395277     6031970       2871541    557110   
7            7  2022          2  1468333     5979214       2723285    556243   
8            8  2022          1  1533240     6073426       2219533    545814   
9            9  2021          4  1657615     5984240       2072236    523107   
10          10  2021          3  1950631     5826257       1566048    492415   
11          11  2021          2  2018587     5641731       1472246    476968   
12          12  2021          1  2175211     5424782       1362976    434548   
13          13  2020          4  2213949     5371824       1311199    421554   
14          14  2020          3  2263889     5259351       1170879    413259   
15          15  2020          2  2295533     5005115       1106725    376667   
16          16  2020          1  2299457     4903674       1047817    352333   
17          17  2019          4  2414575     4883869        941295    340144   
18          18  2019          3  2909805     4800091        852221    338241   
19          19  2019          2  2990457     4635575        760794    308027   
20          20  2019          1  3132789     4435439        637099    276419   
21          21  2018          4  3329027     4313831        388013    264299   
22          22  2018          3  3521447     4238171        247366    246181   
23          23  2018          2  3557216     4038141        226285    194267   
24          24  2018          1  3574294     3981129        217460    165300   
25          25  2017          4  3584311     3818157        211425    155775   
26          26  2017          3  3622575     3670221        193964    106443   
27          27  2017          2  3723518     3383434        180777     85813   
28          28  2017          1  3708898     3276251        180146     84813   
29          29  2016          4  3722794     3210602        178070     85119   
30          30  2016          3  3776442     3124855        169898     81455   
31          31  2016          2  3782085     3035272        167788     85452   
32          32  2016          1  3792493     2806359        164371     85370   
33          33  2015          4  3803024     2898226        139187     85726   
34          34  2015          3  3788696     2840203        162663     84530   
35          35  2015          2  3767821     2756294        150839     82077   
36          36  2015          1  3756153     2668248        168188     79098   
37          37  2014          4  3764038     2536219        149682     76984   
38          38  2014          3  3714764     2569868        155494     85096   
39          39  2014          2  3708882     2461670        149363     72405   
40          40  2014          1  3697066     2407330        150323     70749   

     Otros     Total        Periodo  
0   257941  11633156   Ene-Mar 2024  
1   194796  11547661   Oct-Dic 2023  
2   200027  11406908  Jul-Sept 2023  
3   202428  11251897   Abr-Jun 2023  
4   214838  11241895   Ene-Mar 2023  
5   221740  11208114   Oct-Dic 2022  
6   235230  11091128  Jul-Sept 2022  
7   231609  10958684   Abr-Jun 2022  
8   251996  10624009   Ene-Mar 2022  
9   252596  10489794   Oct-Dic 2021  
10  250190  10085541  Jul-Sept 2021  
11  253552   9863084   Abr-Jun 2021  
12  250455   9647972   Ene-Mar 2021  
13  253036   9571562   Oct-Dic 2020  
14  248821   9356199  Jul-Sept 2020  
15  247016   9031056   Abr-Jun 2020

In [241]:
Totales_Accesos_Por_Tecnología.Periodo.value_counts()

Periodo
Ene-Mar 2024     1
Oct-Dic 2023     1
Jul-Sept 2023    1
Abr-Jun 2023     1
Ene-Mar 2023     1
Oct-Dic 2022     1
Jul-Sept 2022    1
Abr-Jun 2022     1
Ene-Mar 2022     1
Oct-Dic 2021     1
Jul-Sept 2021    1
Abr-Jun 2021     1
Ene-Mar 2021     1
Oct-Dic 2020     1
Jul-Sept 2020    1
Abr-Jun 2020     1
Ene-Mar 2020     1
Oct-Dic 2019     1
Jul-Sept 2019    1
Abr-Jun 2019     1
Ene-Mar 2019     1
Oct-Dic 2018     1
Jul-Sept 2018    1
Abr-Jun 2018     1
Ene-Mar 2018     1
Oct-Dic 2017     1
Jul-Sept 2017    1
Abr-Jun 2017     1
Ene-Mar 2017     1
Oct-Dic 2016     1
Jul-Sept 2016    1
Abr-Jun 2016     1
Ene-Mar 2016     1
Oct-Dic 2015     1
Jul-Sept 2015    1
Abr-Jun 2015     1
Ene-Mar 2015     1
Oct-Dic 2014     1
Jul-Sept 2014    1
Abr-Jun 2014     1
Ene-Mar 2014     1
Name: count, dtype: int64

---

###
###

---

###
###

# Hoja 11

In [242]:
Totales_Accesos_por_velocidad = pd.read_csv("../../data/converted/Totales Accesos por velocidad.csv")
Totales_Accesos_por_velocidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 12 columns):
 #   Column                   Non-Null Count  Dtype
---  ------                   --------------  -----
 0   Unnamed: 0               41 non-null     int64
 1   Año                      41 non-null     int64
 2   Trimestre                41 non-null     int64
 3   Hasta 512 kbps           41 non-null     int64
 4   Entre 512 Kbps y 1 Mbps  41 non-null     int64
 5   Entre 1 Mbps y 6 Mbps    41 non-null     int64
 6   Entre 6 Mbps y 10 Mbps   41 non-null     int64
 7   Entre 10 Mbps y 20 Mbps  41 non-null     int64
 8   Entre 20 Mbps y 30 Mbps  41 non-null     int64
 9   Más de 30 Mbps           41 non-null     int64
 10  OTROS                    41 non-null     int64
 11  Total                    41 non-null     int64
dtypes: int64(12)
memory usage: 4.0 KB


In [243]:
Totales_Accesos_por_velocidad

Unnamed: 0   Año  Trimestre  Hasta 512 kbps  Entre 512 Kbps y 1 Mbps  \
0            0  2024          1           28801                    69355   
1            1  2023          4           29708                    71742   
2            2  2023          3           30274                    80609   
3            3  2023          2           30827                    79405   
4            4  2023          1           31437                    82048   
5            5  2022          4           31981                    85668   
6            6  2022          3           33013                    96727   
7            7  2022          2           33667                    99498   
8            8  2022          1           34890                   104840   
9            9  2021          4           41262                    28521   
10          10  2021          3           40174                    41437   
11          11  2021          2           40172                    42024   
12          12  2021          1           39487                    41674   
13          13  2020          4           39510                    42185   
14          14  2020          3           41038                    44005   
15          15  2020          2           41038                    48690   
16          16  2020          1           42550                    56170   
17          17  2019          4           38272                    28980   
18          18  2019          3          241713                    80599   
19          19  2019          2          202790                    95980   
20          20  2019          1          166573                   195706   
21          21  2018          4          138740                   201196   
22          22  2018          3            5675                   220589   
23          23  2018          2            5972                   219467   
24          24  2018          1            7309                   215406   
25          25  2017          4           15041                   221474   
26          26  2017          3           19022                   217997   
27          27  2017          2           22366                   222766   
28          28  2017          1           20653                   230159   
29          29  2016          4           20724                   233398   
30          30  2016          3           20104                   294413   
31          31  2016          2           44008                   384221   
32          32  2016          1           30428                   404810   
33          33  2015          4           34243                   427394   
34          34  2015          3           35030                   455777   
35          35  2015          2           40723                   500175   
36          36  2015          1           41158                   516919   
37          37  2014          4           44075                   554749   
38          38  2014          3           37430                   608018   
39          39  2014          2           41064                   656408   
40          40  2014          1           52684                   687619   

    Entre 1 Mbps y 6 Mbps  Entre 6 Mbps y 10 Mbps  Entre 10 Mbps y 20 Mbps  \
0                  866152                  950930                   672155   
1                  900253                  978108                   697232   
2                  945608                  998725                   720084   
3                  963510                 1028250                   734125   
4                 1001553                 1060321                   729194   
5                 1075272                 1109421                   765015   
6                 1161370                 1151906                   823505   
7                 1193090                 1197030                   856562   
8                 1263273                 1209148                   967508   
9                 1413208      

---

###
###

---

###
###

# Hoja 12

In [244]:
Totales_Dial_BAf = pd.read_csv("../../data/converted/Totales Dial-BAf.csv")
Totales_Dial_BAf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Unnamed: 0        41 non-null     int64 
 1   Año               41 non-null     int64 
 2   Trimestre         41 non-null     int64 
 3   Banda ancha fija  41 non-null     int64 
 4   Dial up           41 non-null     int64 
 5   Total             41 non-null     int64 
 6   Periodo           41 non-null     object
dtypes: int64(6), object(1)
memory usage: 2.4+ KB


In [245]:
Totales_Dial_BAf

Unnamed: 0   Año  Trimestre  Banda ancha fija  Dial up     Total  \
0            0  2024          1          11621363    11793  11633156   
1            1  2023          4          11535868    11793  11547661   
2            2  2023          3          11395115    11793  11406908   
3            3  2023          2          11239781    12116  11251897   
4            4  2023          1          11229459    12436  11241895   
5            5  2022          4          11195668    12446  11208114   
6            6  2022          3          11078691    12437  11091128   
7            7  2022          2          10946248    12436  10958684   
8            8  2022          1          10611390    12619  10624009   
9            9  2021          4          10476933    12861  10489794   
10          10  2021          3          10075184    10357  10085541   
11          11  2021          2           9852702    10382   9863084   
12          12  2021          1           9637956    10016   9647972   
13          13  2020          4           9561546    10016   9571562   
14          14  2020          3           9346183    10016   9356199   
15          15  2020          2           9021040    10016   9031056   
16          16  2020          1           8802435     9991   8812426   
17          17  2019          4           8783053    10128   8793181   
18          18  2019          3           9142891    21793   9164684   
19          19  2019          2           8938427    21754   8960181   
20          20  2019          1           8626323    21812   8648135   
21          21  2018          4           8451841    21814   8473655   
22          22  2018          3           8320223    26812   8347035   
23          23  2018          2           8083533    26911   8110444   
24          24  2018          1           8009981    27072   8037053   
25          25  2017          4           7842778    27444   7870222   
26          26  2017          3           7696594    28349   7724943   
27          27  2017          2           7401134    28530   7429664   
28          28  2017          1           7277314    28540   7305854   
29          29  2016          4           7223128    28545   7251673   
30          30  2016          3           7178340    28610   7206950   
31          31  2016          2           7097604    32475   7130079   
32          32  2016          1           6874704    32652   6907356   
33          33  2015          4           6952289    32542   6984831   
34          34  2015          3           6902267    32801   6935068   
35          35  2015          2           6783279    32909   6816188   
36          36  2015          1           6699714    38018   6737732   
37          37  2014          4           6559172    39324   6598496   
38          38  2014          3           6559264    36007   6595271   
39          39  2014          2           6428329    36139   6464468   
40          40  2014          1           6362108    36290   6398398   

          Periodo  
0    Ene-Mar 2024  
1    Oct-Dic 2023  
2   Jul-Sept 2023  
3    Abr-Jun 2023  
4    Ene-Mar 2023  
5    Oct-Dic 2022  
6   Jul-Sept 2022  
7    Abr-Jun 2022  
8    Ene-Mar 2022  
9    Oct-Dic 2021  
10  Jul-Sept 2021  
11   Abr-Jun 2021  
12   Ene-Mar 2021  
13   Oct-Dic 2020  
14  Jul-Sept 2020  
15   Abr-Jun 2020  
16   Ene-Mar 2020  
17   Oct-Dic 2019  
18  Jul-Sept 2019  
19   Abr-Jun 2019  
20   Ene-Mar 2019  
21   Oct-Dic 2018  
22  Jul-Sept 2018  
23   Abr-Jun 2018  
24   Ene-Mar 2018  
25   Oct-Dic 2017  
26  Jul-Sept 2017  
27   Abr-Jun 2017  
28   Ene-Mar 2017  
29   Oct-Dic 2016  
30  Jul-Sept 2016  
31   Abr-Jun 2016  
32   Ene-Mar 2016  
33   Oct-Dic 2015  
34  Jul-Sept 2015  
35   Abr-Jun 2015  
36   Ene-Mar 2015  
37   Oct-Dic 2014  
38  Jul-Sept 2014  
39   Abr-Jun 2014  
40   Ene-Mar 2014

In [246]:
Totales_Dial_BAf.Periodo.value_counts()

Periodo
Ene-Mar 2024     1
Oct-Dic 2023     1
Jul-Sept 2023    1
Abr-Jun 2023     1
Ene-Mar 2023     1
Oct-Dic 2022     1
Jul-Sept 2022    1
Abr-Jun 2022     1
Ene-Mar 2022     1
Oct-Dic 2021     1
Jul-Sept 2021    1
Abr-Jun 2021     1
Ene-Mar 2021     1
Oct-Dic 2020     1
Jul-Sept 2020    1
Abr-Jun 2020     1
Ene-Mar 2020     1
Oct-Dic 2019     1
Jul-Sept 2019    1
Abr-Jun 2019     1
Ene-Mar 2019     1
Oct-Dic 2018     1
Jul-Sept 2018    1
Abr-Jun 2018     1
Ene-Mar 2018     1
Oct-Dic 2017     1
Jul-Sept 2017    1
Abr-Jun 2017     1
Ene-Mar 2017     1
Oct-Dic 2016     1
Jul-Sept 2016    1
Abr-Jun 2016     1
Ene-Mar 2016     1
Oct-Dic 2015     1
Jul-Sept 2015    1
Abr-Jun 2015     1
Ene-Mar 2015     1
Oct-Dic 2014     1
Jul-Sept 2014    1
Abr-Jun 2014     1
Ene-Mar 2014     1
Name: count, dtype: int64

---

###
###

---

###
###

# Hoja 13

In [247]:
Totales_VMD = pd.read_csv("../../data/converted/Totales VMD.csv")
Totales_VMD.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41 entries, 0 to 40
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Unnamed: 0              41 non-null     int64  
 1   Año                     41 non-null     int64  
 2   Trimestre               41 non-null     int64  
 3   Mbps (Media de bajada)  41 non-null     float64
 4   Trimestre.1             41 non-null     object 
dtypes: float64(1), int64(3), object(1)
memory usage: 1.7+ KB


In [248]:
Totales_VMD

Unnamed: 0   Año  Trimestre  Mbps (Media de bajada)    Trimestre.1
0            0  2024          1              139.150000   Ene-Mar 2024
1            1  2023          4              139.040000   Oct-Dic 2023
2            2  2023          3              129.670000  Jul-Sept 2023
3            3  2023          2              123.950000   Abr-Jun 2023
4            4  2023          1              110.500000   Ene-Mar 2023
5            5  2022          4              106.560000   Oct-Dic 2022
6            6  2022          3               62.460000  Jul-Sept 2022
7            7  2022          2               58.440000   Abr-Jun 2022
8            8  2022          1               55.110000   Ene-Mar 2022
9            9  2021          4               52.340000   Oct-Dic 2021
10          10  2021          3               48.460000  Jul-Sept 2021
11          11  2021          2               45.630000   Abr-Jun 2021
12          12  2021          1               43.110000   Ene-Mar 2021
13          13  2020          4               42.360000   Oct-Dic 2020
14          14  2020          3               40.670000  Jul-Sept 2020
15          15  2020          2               38.320000   Abr-Jun 2020
16          16  2020          1               37.520000   Ene-Mar 2020
17          17  2019          4               28.260000   Oct-Dic 2019
18          18  2019          3               24.180000  Jul-Sept 2019
19          19  2019          2               20.330000   Abr-Jun 2019
20          20  2019          1               17.380000   Ene-Mar 2019
21          21  2018          4               15.950000   Oct-Dic 2018
22          22  2018          3               15.360000  Jul-Sept 2018
23          23  2018          2               13.849540   Abr-Jun 2018
24          24  2018          1               13.221340   Ene-Mar 2018
25          25  2017          4               12.007626   Oct-Dic 2017
26          26  2017          3                8.339048  Jul-Sept 2017
27          27  2017          2                7.159956   Abr-Jun 2017
28          28  2017          1                6.624079   Ene-Mar 2017
29          29  2016          4                6.337384   Oct-Dic 2016
30          30  2016          3                5.938777  Jul-Sept 2016
31          31  2016          2                5.423531   Abr-Jun 2016
32          32  2016          1                5.083702   Ene-Mar 2016
33          33  2015          4                4.990449   Oct-Dic 2015
34          34  2015          3                4.788911  Jul-Sept 2015
35          35  2015          2                4.545381   Abr-Jun 2015
36          36  2015          1                4.350279   Ene-Mar 2015
37          37  2014          4                4.156888   Oct-Dic 2014
38          38  2014          3                3.867265  Jul-Sept 2014
39          39  2014          2                3.758712   Abr-Jun 2014
40          40  2014          1                3.617127   Ene-Mar 2014

In [249]:
Totales_VMD.Trimestre.value_counts()

Trimestre
1    11
4    10
3    10
2    10
Name: count, dtype: int64

---

###
###

---

###
###

# Hoja 14

In [19]:
Velocidad_porc_por_prov = pd.read_csv("../../data/converted/Velocidad % por prov.csv")
Velocidad_porc_por_prov.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 984 entries, 0 to 983
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Unnamed: 0              984 non-null    int64  
 1   Año                     984 non-null    int64  
 2   Trimestre               984 non-null    int64  
 3   Provincia               984 non-null    object 
 4   Mbps (Media de bajada)  984 non-null    float64
dtypes: float64(1), int64(3), object(1)
memory usage: 38.6+ KB


In [20]:
Velocidad_porc_por_prov

Unnamed: 0   Año  Trimestre            Provincia  Mbps (Media de bajada)
0             0  2024          1         Buenos Aires              159.130000
1             1  2024          1      Capital Federal              229.940000
2             2  2024          1            Catamarca               91.450000
3             3  2024          1                Chaco              107.510000
4             4  2024          1               Chubut               20.500000
..          ...   ...        ...                  ...                     ...
979         979  2014          1           Santa Cruz                3.315991
980         980  2014          1             Santa Fe                3.147407
981         981  2014          1  Santiago Del Estero                2.949472
982         982  2014          1     Tierra Del Fuego                3.446888
983         983  2014          1              Tucumán                3.208342

[984 rows x 5 columns]

In [21]:
Velocidad_porc_por_prov.Provincia.value_counts

<bound method IndexOpsMixin.value_counts of 0             Buenos Aires
1          Capital Federal
2                Catamarca
3                    Chaco
4                   Chubut
              ...         
979             Santa Cruz
980               Santa Fe
981    Santiago Del Estero
982       Tierra Del Fuego
983                Tucumán
Name: Provincia, Length: 984, dtype: object>

In [23]:
Velocidad_porc_por_prov.to_parquet("../../data/interim/VMD.parquet")

---

###
###

---

###
###

# Hoja 15

In [253]:
Velocidad_sin_Rangos = pd.read_csv("../../data/converted/Velocidad_sin_Rangos.csv")
Velocidad_sin_Rangos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 625 entries, 0 to 624
Columns: 194 entries, Unnamed: 0 to 10000 Mbps
dtypes: float64(191), int64(2), object(1)
memory usage: 947.4+ KB


In [254]:
Velocidad_sin_Rangos

Unnamed: 0     Año  Trimestre            Provincia     Otros  0,256 Mbps  \
0             0  2024.0        1.0         Buenos Aires  125147.0        18.0   
1             1  2024.0        1.0      Capital Federal       0.0        31.0   
2             2  2024.0        1.0            Catamarca      70.0         0.0   
3             3  2024.0        1.0                Chaco    3816.0         0.0   
4             4  2024.0        1.0               Chubut   16640.0         8.0   
..          ...     ...        ...                  ...       ...         ...   
620         620  2017.0        4.0             Santa Fe   14119.0         0.0   
621         621  2017.0        4.0  Santiago Del Estero       0.0         0.0   
622         622  2017.0        4.0     Tierra Del Fuego       0.0         0.0   
623         623  2017.0        4.0              Tucumán       0.0         0.0   
624         624     NaN        NaN                  NaN       NaN         NaN   

     0,375 Mbps  0,625 Mbps  0,5 Mbps  0,512 Mbps  ...  2048 Mbps  2529 Mbps  \
0           0.0         0.0   25815.0       169.0  ...        0.0        0.0   
1           0.0         0.0     451.0        34.0  ...        0.0        0.0   
2           0.0         0.0      34.0        37.0  ...        0.0        0.0   
3           0.0         0.0     234.0         2.0  ...        0.0        0.0   
4           0.0         0.0      69.0        34.0  ...        0.0        0.0   
..          ...         ...       ...         ...  ...        ...        ...   
620         0.0         0.0    3109.0         0.0  ...        0.0        0.0   
621         0.0         0.0      41.0         0.0  ...        0.0        0.0   
622         0.0         0.0       6.0         0.0  ...        0.0        0.0   
623         0.0         0.0     105.0         0.0  ...        0.0        0.0   
624         NaN         NaN       NaN         NaN  ...        NaN        NaN   

     3072 Mbps  4000 Mbps  5000 Mbps  5125 Mbps  6000 Mbps  7000 Mbps  \
0          0.0        0.0        0.0        0.0        0.0        0.0   
1          0.0        0.0        0.0        0.0        0.0        0.0   
2          0.0        0.0        0.0        0.0        0.0        0.0   
3          0.0        0.0        0.0        0.0        0.0        0.0   
4          0.0        0.0        0.0        0.0        0.0        0.0   
..         ...        ...        ...        ...        ...        ...   
620        0.0        0.0        0.0        0.0        0.0        0.0   
621        0.0        0.0        0.0        0.0        0.0        0.0   
622        0.0        0.0        0.0        0.0        0.0        0.0   
623        0.0        0.0        0.0        0.0        0.0        0.0   
624        NaN        NaN        NaN        NaN        NaN        NaN   

     8000 Mbps  10000 Mbps  
0          0.0         0.0  
1          0.0         0.0  
2          0.0         0.0  
3          0.0         0.0  
4          0.0         0.0  
..         ...         ...  
620        0.0         0.0  
621        0.0         0.0  
622        0.0         0.0  
623        0.0         0.0  
624        NaN         NaN  

[625 rows x 194 columns]

In [255]:
Velocidad_sin_Rangos.describe()

Unnamed: 0          Año   Trimestre          Otros  0,256 Mbps  \
count  625.000000   624.000000  624.000000     624.000000  624.000000   
mean   312.000000  2020.500000    2.500000    9905.139423    3.469551   
std    180.566239     1.907996    1.152846   17973.450593   10.749551   
min      0.000000  2017.000000    1.000000       0.000000    0.000000   
25%    156.000000  2019.000000    1.000000     589.500000    0.000000   
50%    312.000000  2020.500000    2.500000    4467.500000    0.000000   
75%    468.000000  2022.000000    4.000000   13300.750000    1.000000   
max    624.000000  2024.000000    4.000000  125147.000000  102.000000   

       0,375 Mbps  0,625 Mbps       0,5 Mbps  0,512 Mbps     0,75 Mbps  ...  \
count  624.000000  624.000000     624.000000  624.000000    624.000000  ...   
mean     3.966346    1.956731    2236.834936   40.467949   1150.025641  ...   
std     29.632806   14.618851   15902.906210   69.387753   4676.714406  ...   
min      0.000000    0.000000       6.000000    0.000000      0.000000  ...   
25%      0.000000    0.000000      20.750000    0.000000      0.000000  ...   
50%      0.000000    0.000000      47.000000   16.000000      4.000000  ...   
75%      0.000000    0.000000      82.250000   60.000000    318.000000  ...   
max    225.000000  111.000000  238902.000000  609.000000  35185.000000  ...   

        2048 Mbps   2529 Mbps   3072 Mbps   4000 Mbps   5000 Mbps   5125 Mbps  \
count  624.000000  624.000000  624.000000  624.000000  624.000000  624.000000   
mean     6.903846    0.032051    1.818910    0.113782    0.059295    0.006410   
std     76.733746    0.800641   20.254423    1.209195    0.719825    0.079871   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
50%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
75%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
max    860.000000   20.000000  227.000000   15.000000    9.000000    1.000000   

        6000 Mbps   7000 Mbps   8000 Mbps  10000 Mbps  
count  624.000000  624.000000  624.000000  624.000000  
mean     0.054487    0.158654    0.012821    0.073718  
std      0.605151    1.441916    0.159742    0.701533  
min      0.000000    0.000000    0.000000    0.000000  
25%      0.000000    0.000000    0.000000    0.000000  
50%      0.000000    0.000000    0.000000    0.000000  
75%      0.000000    0.000000    0.000000    0.000000  
max      8.000000   18.000000    2.000000    9.000000  

[8 rows x 193 columns]

Como realice anteriormente, elimino las filas en las que no aparecen nombres de provincias

In [256]:
Velocidad_sin_Rangos = Velocidad_sin_Rangos.dropna(subset='Provincia')

In [257]:
Velocidad_sin_Rangos = Velocidad_sin_Rangos.fillna(0)

In [258]:
Velocidad_sin_Rangos.Provincia.value_counts()

Provincia
Buenos Aires           26
Capital Federal        26
Catamarca              26
Chaco                  26
Chubut                 26
Córdoba                26
Corrientes             26
Entre Ríos             26
Formosa                26
Jujuy                  26
La Pampa               26
La Rioja               26
Mendoza                26
Misiones               26
Neuquén                26
Río Negro              26
Salta                  26
San Juan               26
San Luis               26
Santa Cruz             26
Santa Fe               26
Santiago Del Estero    26
Tierra Del Fuego       26
Tucumán                26
Name: count, dtype: int64